python/cudf/udf_cpp/groupby/function.cu

```diff
@@ -0,0 +1,623 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cfloat>
+#include <cstdint>
+
+using size_type = int;
+
+// double atomicAdd
+__device__ __forceinline__ double atomicAdds(double* address, double val)
```

Comment:
All of these custom atomics can/should be replaced with `cuda::atomic_ref` https://godbolt.org/z/r3xGxM8zj

Reply:
Done.